### PR TITLE
Add Python 3.7 support. 

### LINKED ISSUE #214: Add support for Python 3.7

### LINKED ISSUE #215: Fix deprecations seen during testing
When zope.testrunner started doing the recommended thing and enabling deprecation warnings, we started showing some:
```
//ZODB/src/ZODB/scripts/tests/test_doc.py:22: DeprecationWarning: invalid escape sequence \.
  '[0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2}:[0-9]{2}\.[0-9]+'),
//ZODB/src/ZODB/scripts/tests/test_doc.py:32: DeprecationWarning: invalid escape sequence \(
  "length=[0-9]+ offset=[0-9]+ \(\+23\)"),
//ZODB/src/ZODB/scripts/tests/test_doc.py:36: DeprecationWarning: invalid escape sequence \(
  "length=[0-9]+ offset=4 \(\+48\)"),
//ZODB/src/ZODB/tests/testConnection.py:39: DeprecationWarning: invalid escape sequence \(
  (re.compile("set\(\[\]\)"), r"set()"),
Running zope.testrunner.layer.UnitTests tests:
  Set up zope.testrunner.layer.UnitTests in 0.000 seconds.
//ZODB/src/ZODB/tests/testConnection.py:201: DeprecationWarning: Please use assertTrue instead.
  self.assert_(record.exc_info)
//ZODB/src/ZODB/Connection.py:1331: DeprecationWarning: _extension is deprecated, use extension
  DeprecationWarning)
```

### PR DESCRIPTION
**Pull request by jamadden**
And fix all the deprecations it reports.

Fixes #214 and fixes #215 (well there's one remaining deprecation in 'manuel')

Based on #217 so that all tests pass.

### CODE REVIEW

File: src/ZODB/Connection.py [hunk 1]
```diff
@@ -1328,7 +1328,7 @@ def note(self, text): # for tests
     @property
     def _extension(self):
         warnings.warn("_extension is deprecated, use extension",
-                      DeprecationWarning)
+                      DeprecationWarning, stacklevel=2)
```
Comment (tseaver) [inline src/ZODB/Connection.py]: This change looks unrelated to the PR.

Comment (jamadden) [reply src/ZODB/Connection.py]: In order to solve the DeprecationWarning, I needed to know where it was coming from. Without specifying the stacklevel, the warning appears to be coming from this code, not from the place that's actually triggering the warning.

**Comment by jamadden** [comment]
Rebased on master and re-pushed since there was a conflict.

File: src/ZODB/Connection.py (hunk 1 above)
Comment (tseaver) [reply src/ZODB/Connection.py]: OIC

File: tox.ini [hunk 2]
```diff
@@ -16,7 +16,7 @@ commands =
 # Run unit tests first.
     zope-testrunner -u --test-path=src []
 # Only run functional tests if unit tests pass.
-    zope-testrunner -f --test-path=src []
+    zope-testrunner -f -j5 --test-path=src []
```
Comment (tseaver) [inline tox.ini]: Yay!

**Comment by jamadden** [comment]
Thanks! We've got a [green build on appveyor](https://ci.appveyor.com/project/jimfulton/zodb/build/1.0.155) plus all the builds on travis, but appveyor is backed up with PRs to build so it's going to be awhile before we get green checkmarks here, so I'm going to merge (which will cause some of those queued builds to go away).

**Comment by icemac** [comment]
@tseaver @jimfulton Can we have a new release?

**Comment by tseaver** [comment]
@icemac See #225.
